clang/include/clang/Driver/Driver.h

```diff
@@ -718,6 +718,28 @@ class Driver {
     return IsOffload ? OffloadLTOMode : LTOMode;
   }
 
+  /// FPGA Emulation Mode.  By default, this is true due to the fact that
+  /// an external option setting is required to target hardware.
+  enum DeviceMode {
+    FPGAHWMode,
+    FPGAEmulationMode
+  } OffloadCompileMode = FPGAEmulationMode;
```

Comment:
We should extend this a bit more.  By default, there shouldn't be a specific mode set, so we can introduce some default 'unset' value.  For example, if one were to use `'-fsycl-targets=spir64_x86_64` this device target has nothing to do with FPGA, but we would have `FPGAEmulationMode` set anyway.  Having an 'unset' type of default value can give use that extra bit of control.

Reply:
`FPGAEmulationMode` is set as the default value in the old code. It is not being set from anywhere else.
If we were to remove the default `FPGAEmulationMode` value, where should we set this value from?

Reply:
The default would be a new value, maybe `UnsetDeviceMode` or something like that.